tests/tools/test_var.py

```diff
@@ -51,3 +51,64 @@ def test_large_epsilon_axis(self):
 
         for i in range(res.shape[0]):
             self.assertAlmostEqual(res[i], res_dp[i], delta=0.01)
+
+    def test_nan(self):
+        a = np.random.random((5, 5))
+        a[2, 2] = np.nan
+
+        res = var(a, range=1)
+        self.assertTrue(np.isnan(res))
+
+
+class TestNanVar(TestCase):
+    def test_not_none(self):
+        mech = nanvar
+        self.assertIsNotNone(mech)
+
+    def test_no_params(self):
+        a = np.array([1, 2, 3])
+        with self.assertWarns(PrivacyLeakWarning):
+            res = nanvar(a)
+        self.assertIsNotNone(res)
+
+    def test_no_epsilon(self):
+        a = np.array([1, 2, 3])
+        self.assertIsNotNone(nanvar(a, range=1))
+
+    def test_no_range(self):
+        a = np.array([1, 2, 3])
+        with self.assertWarns(PrivacyLeakWarning):
+            nanvar(a, epsilon=1)
+
+    def test_negative_range(self):
+        a = np.array([1, 2, 3])
+        with self.assertRaises(ValueError):
+            nanvar(a, epsilon=1, range=-1)
+
+    def test_missing_range(self):
+        a = np.array([1, 2, 3])
+        with self.assertWarns(PrivacyLeakWarning):
+            res = nanvar(a, 1, None)
+        self.assertIsNotNone(res)
+
+    def test_large_epsilon(self):
+        a = np.random.random(1000)
+        res = float(np.var(a))
+        res_dp = nanvar(a, epsilon=1, range=1)
+
+        self.assertAlmostEqual(res, res_dp, delta=0.01)
+
+    def test_large_epsilon_axis(self):
+        a = np.random.random((1000, 5))
+        res = np.var(a, axis=0)
+        res_dp = nanvar(a, epsilon=1, range=1, axis=0)
+
+        for i in range(res.shape[0]):
+            self.assertAlmostEqual(res[i], res_dp[i], delta=0.01)
+
+    def test_nan(self):
+        a = np.random.random((5, 5))
+        a[2, 2] = np.nan
+
+        res = nanvar(a, range=1)
+        self.assertFalse(np.isnan(res))
```

Comment:
I start seeing a pattern here

Reply:
Thanks for the suggestion, it has been fixed in https://github.com/IBM/differential-privacy-library/pull/13/commits/81161c91ae332cded7c897818f9794a1d49373da